packages/yarnpkg-parsers/sources/grammars/shell.pegjs

```diff
@@ -94,10 +94,10 @@ PlainStringSegment
   / text:PlainStringText { return { type: `text`, text } }
 
 SglQuoteStringText
-  = chars:('\\' c:. { return c } / [^'])* { return chars.join(``) }
+  = chars:('\\' c:[^u0'] { return c } / [^'])* { return chars.join(``) }
 
 DblQuoteStringText
-  = chars:('\\' c:. { return c } / [^$"])+ { return chars.join(``) }
+  = chars:('\\' c:[^u0'] { return c } / [^$"])+ { return chars.join(``) }
```

Comment:
That doesn't seem right - rather, the fix would be to add support for unicode syntax to the parser. Ie something like this:

```
  = chars:('\\u' val:[0-9]+ { return String.fromCharCode(parseInt(val, 16)) } / '\\' c:. { return c } / [^$"])+ { return chars.join(``) }
```

Reply:
I know that is not the correct one, but if parse unicode here, it should be back in the code-gen step, i.e. the function stringifyArgumentSegment should change either. So I think it is not necessary to make a heavy parser here.

Reply:
Does this project require a full bash parser? There are still some bash syntaxes that have not been implemented. like backquote `echo "echo  test"`. it works with yarn 1.x and npm.